src/runtime_src/core/common/api/command.h

```diff
@@ -64,6 +64,12 @@ class command : public std::enable_shared_from_this<command>
   virtual void
   notify(ert_cmd_state) const = 0;
 
+  /**
+  * get_hw_context() - get hw context of command buffer
+  */
+  virtual xrt::hw_context
+  get_hw_context() const = 0;
```

Comment:
```suggestion
  virtual xcl_hwctx_handle
  get_hwctx_handle() const = 0;
```
I think it would be cheaper if we use `get_hwctx_handle()`. This will avoid reference counting magic, which while cheap is not for free.

Reply:
fixed